substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1203,11 +1204,30 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
         }
 
         int exitStatus = 1;
+        File argsFile = null;
         try {
+            if (OS.getCurrent() == OS.WINDOWS) {
+                int totalLength = command.stream().mapToInt(s -> s.length() + 1).sum();
+                if (totalLength >= WINDOWS_MAX_COMMAND_LENGTH) {
```

Comment:
If we make this the default for Java > 8 then we don't need to make this code Windows specific.

Reply:
Yes, I said the same in the PR description. Is that what you want me to do? Just remove the platform check and make this the default for all platforms?

Reply:
If we are confident after extensive testing that it causes no drawbacks then we should make that the default for Java > 8 (independent of the platform).

Reply:
Please change the sources to use
```
        if (config.useJavaModules()) {
            /* For Java > 8 we use an argument file to pass the options to the builder */
```
instead of `if (Platform.includedIn(Platform.WINDOWS.class))` ... `if (totalLength >= WINDOWS_MAX_COMMAND_LENGTH) {`. Then I can run this PR in our CI to see if the argument file approach causes any trouble.

Reply:
Done